pkg/api/deploymentapi/get_resource_ref_id.go

```diff
@@ -0,0 +1,105 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package deploymentapi
+
+import (
+	"fmt"
+
+	"github.com/elastic/cloud-sdk-go/pkg/api"
+	"github.com/elastic/cloud-sdk-go/pkg/util"
+	"github.com/elastic/cloud-sdk-go/pkg/util/ec"
+)
+
+// PopulateRefIDParams is consumed by PopulateRefID.
+type PopulateRefIDParams struct {
+	*api.API
+
+	DeploymentID string
+	Kind         string
+
+	// RefID string pointer where the value is persisted.
+	RefID *string
+}
+
+// GetKindRefID obtains a resource kind RefID. If the kind is not supported
+// an error is returned.
+func GetKindRefID(params GetResourceParams) (string, error) {
```

Comment:
with the new `PopulateRefID()` function does this one still need to be public?

Reply:
It is because it's used on the `depresoruceapi` package, I was thinking about reviewing the usages of the ref-id auto-discovery and change what's necessary in a follow-up PR.

I'll open a ticket about it.

Reply:
sounds good!